operator/pkg/resources/configmap.go

```diff
@@ -264,3 +264,35 @@ func (r *ConfigMapResource) SetAnnotationForCluster(
 	}
 	return r.Update(ctx, &existing)
 }
+
+// ConfigTemplater is an interface for objects that can provide template data for ConfigMap creation
+type ConfigTemplater interface {
+	Templates() (map[string]string, error)
```

Comment:
@andrewstucki and I don't exactly see eye to eye on this but I _really_ like closures (i.e. `func() (map[string]string, error)` ) for uses like this.

My rule of thumb when it comes to interface vs closure is that statefulness should be represented as an interface. A closure _should_ indicate statelessness.

For example, `io.Reader` is stateful. The variable that implements the interface is just as important as having access to the `Read` method itself.

This function, on the other hand, seems like it will return the same values every time you call it. Therefore the object containing it really doesn't matter to the consumer.